components/list/list-item-drag-mixin.js

```diff
@@ -158,9 +432,8 @@ export class NewPositionEventDetails {
 	 */
 	reorder(list, {announceFn, keyFn}) {
```

Comment:
The reorder logic doesn't seem like it should belong to the event details.  After all, these is just positioning event info.  While it is nice that it works on a generic list, I think this logic better resides in a list component/mixin.

Reply:
So all the rendering is going to be handled by the hypermedia components that are using this component. The functionality of the list is basically send events up to say this has move above/below another item in the list.

So any consumer will have a list of items. When we tell them, hey this item should move here. They would then have to grab their list and reorder it based on the information. Which they would be like, that is easy enough I'll just write a private method that does it. So we would then get a lot of different ways on reordering a list. It is fun to write real world interview questions but I figured why not just send up a method that does it for them. So then we only have one piece of code the reorders a list.

This also solve another issue where we need to announce what is moving and where it is going for screen readers. Our list component is dumb and has no knowledge of its contents. Each list item doesn't even know what position in the list it is in. So instead of relaying on every implementation of the list to do things right with accessibility. I figured let's make it as easy as possible by giving them ability to run the above method to announce the change where the information required is known.

Also, I plan on making another change where destination could be move up 1 or move down 1 (which would easily translate to move by n and move down by n) So all I would have to do is update this method and both types of destinations would work easily and the user wouldn't have to worry about which type of destination is sent up.

If it is a performance worry. We could always export the method instead of sending up the class and the new method would just take the event as a parameter. 

I would still prefer to keep as part of the detail class information since it more organized and easier to see that it exists for someone using it. 

As for the logic living in the mixin or list component sadly the information required to run this method is not something either has access too.

Reply:
Yeah I am all for writing this logic once and only once.  I most definitely don't want to see consumers having to write this sort of logic, or the complex layout logic we have.  It all needs to be very simple for consumers to use.

> As for the logic living in the mixin or list component sadly the information required to run this method is not something either has access too.

Hmmm, this is unfortunate.  A details/info class I usually expect to be just that.  I'll give it another look.